### ISSUE TITLE
`bugprone-standalone-empty` false positive when returning the result

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#59517]
```cpp
#include <string>

struct A
{
    bool empty() const
    {
        return s.empty();
    }
    std::string s;
};
```

```console
<source>:7:16: warning: ignoring the result of 'empty()' [bugprone-standalone-empty]
        return s.empty();
               ^
```

https://godbolt.org/z/v7rxqY1G7

**Comment by v1nh1shungry** [comment]
Proposed fix: https://reviews.llvm.org/D141107

**Comment by cjdb** [comment]
Fixed. Thanks @v1nh1shungry!
